### PR TITLE
Rename deprecated method

### LINKED ISSUE #2528: UserWarning regarding the RSquare metric when used in Keras model
\Anaconda3\lib\site-packages\tensorflow\python\keras\metrics.py:254: UserWarning: Metric RSquare implements a `reset_states()` method; rename it to `reset_state()` (without the final "s"). The name `reset_states()` has been deprecated to improve API consistency.
  warnings.warn('Metric %s implements a `reset_states()` method; rename it '

Tensorflow 2.5.0
Keras 2.5.0
tensorflow_addos 0.13.0

### PR DESCRIPTION
**Pull request by RyanGoslingsBugle**
# Description

Brief Description of the PR: Rename deprecated reset_states() method to reset_state(). Update documentation and test cases.

Fixes #2528

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [x] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [x] This PR addresses an already submitted issue for TensorFlow Addons
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*  All metrics operate as expected, minor refactor

### CODE REVIEW

**Comment by RyanGoslingsBugle** [comment]
@googlebot I signed it!

**Comment by bhack** [comment]
Thanks for the PR, you need to fix linting.

**Comment by RyanGoslingsBugle** [comment]
@bhack Cheers, should be fixed in d3fb8e2a07574b35c0bf9ab34561c6547cc52064

**Comment by bhack** [comment]
You need to conditonally handle this as old version tests are failing.

**Review by bhack — COMMENTED** [review]
Why we have some large diffs?

**Comment by RyanGoslingsBugle** [comment]
@bhack something went wrong with the auto-generated changes that came from the pre-commit hook, I've reverted and made them manually

**Review by AakashKumarNain — APPROVED** [review]

File: tensorflow_addons/metrics/cohens_kappa.py [hunk 1]
```diff
@@ -254,11 +254,16 @@ def get_config(self):
...
         """Resets all of the metric state variables."""
 
         for v in self.variables:
             K.set_value(
                 v,
                 np.zeros((self.num_classes, self.num_classes), v.dtype.as_numpy_dtype),
             )
+
+    def reset_states(self):
+        # Backwards compatibility alias of `reset_state`. New classes should
```
Comment (bhack) [inline tensorflow_addons/metrics/cohens_kappa.py:267]: Can you add the version `< <TF version>` in the comment? It will be a good remainder to know when we could remove it.

**Review by bhack — CHANGES_REQUESTED** [review]

File: tensorflow_addons/metrics/cohens_kappa.py (hunk 1 above)
Comment (RyanGoslingsBugle) [reply tensorflow_addons/metrics/cohens_kappa.py:267]: Resolved in 9d8e660

**Review by bhack — APPROVED** [review]

**Comment by bhack** [comment]
Thanks

**Comment by lucasdavid** [comment]
This closes #2490 too.
